### PR TITLE
Add birthday attribute and tests

### PR DESCRIPTION
**Pull request by Dban1**
-implemented Birthday and BirthdayTest

A valid birthday satisfies the following:
-in the format "DD-MM-YYYY"
-input date is earlier than the current date

BirthdayTest includes a hard-coded test case for a birthday on 1st January 1980.

### CODE REVIEW

**Comment by Yonggiee** [comment]
fail test coverage, add more tests or uwuhan

**Comment by Dban1** [comment]
Improved coverage.

**Review by AaronCQL — DISMISSED** [review]
LGTM

File: src/main/java/seedu/address/model/client/Birthday.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    public static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+    public final LocalDate value;
+
+    /**
+     * Constructs a {@code Birthday}.
+     *
+     * @param birthday A valid birth date in the form DD-MM-YYYY.
+     */
+    public Birthday(String birthday) {
+        checkArgument(isValidBirthday(birthday), MESSAGE_CONSTRAINTS);
```
Comment (AaronCQL) [inline src/main/java/seedu/address/model/client/Birthday.java:29]: I think we should include the `requireNonNull(birthday)` here.

File: src/test/java/seedu/address/model/client/BirthdayTest.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                .format(DateTimeFormatter.ofPattern("dd-MM-yyyy")))); //birthday 2 months after current date
+        assertFalse(Birthday.isValidBirthday(LocalDate.now()
+                .format(DateTimeFormatter.ofPattern("dd-MM-yyyy")))); // birthday today
+
+        // valid birthday
+        assertTrue(Birthday.isValidBirthday("01-01-1980")); // birthday in 01-01-1980
+        assertTrue(Birthday.isValidBirthday("01-01-1970")); //birthday in 01-01-1979
+
+        assertTrue(new Birthday("01-01-1980").hashCode() == new Birthday("01-01-1980").hashCode());
+        assertTrue(new Birthday("01-01-1980").toString().equals("01-01-1980") == true);
```
Comment (ziyingli) [inline src/test/java/seedu/address/model/client/BirthdayTest.java:54]: abstract awayy

**Review by ziyingli — APPROVED** [review]
LGTM

**Pull request merged**